utils/appliance/__init__.py

```diff
@@ -2097,3 +2135,12 @@ def get_or_create_current_appliance():
     return stack.top
 
 current_appliance = LocalProxy(get_or_create_current_appliance)
+
+
+class CurrentAppliance(object):
+    def __init__(self):
+        self.ca = None
+
+    def __get__(self, instance, owner):
+        self.ca = get_or_create_current_appliance()
```

Comment:
So I tried this
```python
class Test(object):
    app = CurrentAppliance()

test = Test()
with IPAppliance(10.1.1.1):
    print Test.app
    with IPAppliance(10.2.2.2):
        print Test.app
```

and got

```
INFO:cfme:Pushed appliance 10.1.1.1 on stack (was 10.x.x.x before) 
IPAppliance('10.1.1.1')
INFO:cfme:Pushed appliance 10.2.2.2 on stack (was 10.1.1.1 before) 
IPAppliance('10.2.2.2')
INFO:cfme:Popped appliance 10.2.2.2 from the stack (now there is 10.1.1.1)
INFO:cfme:Popped appliance 10.1.1.1 from the stack (now there is 10.x.x.x)
```

That looks to me like the appliance that the class would use would change when inside the stack which is the expected behaviour.